ModularTegustation/tegu_items/gadgets/unpowered.dm

```diff
@@ -273,43 +279,101 @@
 	attack_verb_simple = list("attack", "bash", "batter", "bludgeon", "whack")
 	toolspeed = 1
 	armor = list(MELEE = 0, BULLET = 0, LASER = 0, ENERGY = 0, BOMB = 0, BIO = 0, RAD = 0, FIRE = 100, ACID = 100)
-	var/mode = 1
+	var/mode = RAK_HP_MODE
 
 /obj/item/safety_kit/attack_self(mob/user)
 	if(!clerk_check(user))
 		to_chat(user,"<span class='warning'>You don't know how to use this.</span>")
 		return
+	ChangeMode(user)
+	return
+
+/obj/item/safety_kit/attack_obj(obj/O, mob/living/user)
+	if(!istype(O, /obj/machinery/regenerator))
+		return ..()
+	Augment(O, user)
+
+/obj/item/safety_kit/proc/Augment(obj/machinery/regenerator/R, mob/living/user)
+	. = FALSE
+	if(!clerk_check(user))
+		to_chat(user,"<span class='warning'>You don't know how to use this.</span>")
+		return
+	if(R.modified)
+		to_chat(user, "<span class='notice'>The [R] is already modified.</span>")
+		return
+	to_chat(user, "<span class='notice'>You begin tinkering with the [R].</span>")
+	if(!do_after(user, 2.5 SECONDS, R, extra_checks = CALLBACK(src, .proc/ModifiedCheck, R)))
+		to_chat(user, "<span class='spider'>Your work has been interrupted!</span>")
+		return
+	R.modified = TRUE
```

Comment:
Add an extra check after do_after to make sure it was not modified while you were in the process.

Reply:
Ah, wait, that's what extra check does, I suppose.